cpu/native/socket_zep/socket_zep.c

```diff
@@ -0,0 +1,441 @@
+static int _set(netdev_t *netdev, netopt_t opt, void *value, size_t value_len);
+
+static const netdev_driver_t socket_zep_driver = {
+    .send = _send,
+    .recv = _recv,
+    .init = _init,
+    .isr = _isr,
+    .get = _get,
+    .set = _set,
+};
+
+static size_t _zep_hdr_fill_v2_data(socket_zep_t *dev, zep_v2_data_hdr_t *hdr,
+                                    size_t payload_len)
+{
+    struct timeval tv;
+
+    real_gettimeofday(&tv, NULL);
+    hdr->hdr.version = 2;
+    hdr->type = ZEP_V2_TYPE_DATA;
+    hdr->chan = dev->netdev.chan;
+    hdr->dev = byteorder_htons((uint16_t)((((intptr_t)dev)) & 0xffff));
+    hdr->lqi_mode = 1;
+    hdr->lqi_val = 0xff;                /* TODO: set */
+    hdr->time.seconds = byteorder_htonl(tv.tv_sec + _UNIX_NTP_ERA_OFFSET);
+    hdr->time.fraction = byteorder_htonl((tv.tv_usec * 1000000) / 232);
+    hdr->seq = byteorder_htonl(dev->seq);
+    memset(hdr->resv, 0, sizeof(hdr->resv));
+    hdr->length = payload_len;
+
+    return sizeof(zep_v2_data_hdr_t);
+}
+
+static inline size_t _zep_hdr_fill(socket_zep_t *dev, zep_hdr_t *hdr,
+                                   size_t payload_len)
+{
+    hdr->preamble[0] = 'E';
+    hdr->preamble[1] = 'X';
+
+    /* keep possibility for ZEPv1 open */
+    return _zep_hdr_fill_v2_data(dev, (zep_v2_data_hdr_t *)hdr,
+                                 payload_len);
+}
+
+static size_t _prep_vector(socket_zep_t *dev, const struct iovec *vector,
+                           unsigned n, struct iovec *out)
+{
+    size_t bytes = 0;
+    dev->chksum_buf = 0;
+
+    for (unsigned i = 0; i < n; i++) {
+        bytes += vector[i].iov_len;
+    }
+    bytes += sizeof(uint16_t); /* FCS field */
+    out[0].iov_base = &dev->snd_hdr_buf;
+    out[0].iov_len = _zep_hdr_fill(dev, out[0].iov_base, bytes);
+    for (unsigned i = 0; i < n; i++) {
+        /* discard const qualifier, we won't change anything. Promise! */
+        out[i + 1].iov_base = vector[i].iov_base;
+        out[i + 1].iov_len = vector[i].iov_len;
+        dev->chksum_buf = ucrc16_calc_le(out[i + 1].iov_base, out[i + 1].iov_len,
+                                         UCRC16_CCITT_POLY_LE, dev->chksum_buf);
+    }
+    dev->chksum_buf = byteorder_btols(byteorder_htons(dev->chksum_buf)).u16;
+    out[n + 1].iov_base = &dev->chksum_buf;
+    out[n + 1].iov_len = sizeof(uint16_t);
+    return bytes;
+}
+
+static int _send(netdev_t *netdev, const struct iovec *vector, unsigned n)
+{
+    socket_zep_t *dev = (socket_zep_t *)netdev;
+    struct iovec v[n + 2];
+    size_t bytes;
+    int res;
+
+    assert((dev != NULL) && (dev->sock_fd != 0));
+    bytes = _prep_vector(dev, vector, n, v);
+    DEBUG("socket_zep::send(%p, %p, %u)\n", (void *)netdev, (void *)vector, n);
+    /* simulate TX_STARTED interrupt */
+    if (netdev->event_callback) {
+        dev->last_event = NETDEV_EVENT_TX_STARTED;
+        netdev->event_callback(netdev, NETDEV_EVENT_ISR);
+        thread_yield();
```

Comment:
Ok 'good' to hear. I didn't get it running either. Please ping me once fixed. Regarding `thread_yield()` I am wondering to which thread you want to yield...